src/v/utils/external_process.cc

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "external_process.h"
+
+#include <seastar/core/seastar.hh>
+#include <seastar/core/timed_out_error.hh>
+#include <seastar/core/when_all.hh>
+#include <seastar/core/with_timeout.hh>
+#include <seastar/util/later.hh>
+
+namespace external_process {
+ss::future<external_process> external_process::create_external_process(
+  std::vector<ss::sstring> command,
+  std::optional<std::vector<ss::sstring>> env) {
+    vassert(
+      !command.empty(),
+      "Should not be passed an empty command list to "
+      "create_external_process");
+
+    ss::experimental::spawn_parameters params;
+    auto path = std::filesystem::path(command[0]);
+    params.argv = std::move(command);
+    if (env) {
+        params.env = std::move(env).value();
+    }
+
+    auto p = co_await ss::experimental::spawn_process(path, params);
+
+    co_return external_process(
+      std::move(p), std::move(path), std::move(params));
+}
+
+external_process::external_process(external_process&& rhs) noexcept
+  : _process(std::move(rhs._process))
+  , _path(std::move(rhs._path))
+  , _params(std::move(rhs._params))
+  , _completed(rhs._completed)
+  , _terminate_mutex(std::move(rhs._terminate_mutex))
+  , _wait_mutex(std::move(rhs._wait_mutex))
+  , _stdout_consumer(std::move(rhs._stdout_consumer))
+  , _stderr_consumer(std::move(rhs._stderr_consumer))
```

Comment:
I think we need to explicitly `std::exchange` with `nullopt` to make sure the assert in the destructor doesn't fire...